crates/driver/src/domain/eth/mod.rs

```diff
@@ -195,6 +195,12 @@ impl TokenAmount {
                 .into(),
         )
     }
+
+    pub fn checked_sub(self, other: TokenAmount) -> Option<TokenAmount> {
```

Comment:
why not reusing num::CheckedSub

Reply:
Mostly to avoid doing `value_a.0.checked_sub(value_b.0)`, and having `value_a.checked_sub(value_b)`.

Reply:
You could then also implement `num::CheckSub` on the wrapper type, no?

Reply:
@MartinquaXD what do you mean? it is implemented in the wrapper type `TokenAmount` :thinking:

Reply:
Yes, but Dusan was referring to implementing it like:
```rust
impl num::CheckedSub for TokenAmount {
//
}
```
Instead of having it in the regular `impl TokenAmount` block to possibly get some nicer ergonomics elsewhere.

Reply:
Ah, I see what you mean :+1: 

> Also the checked_sub function on TokenAmount should check that the tokens are actually the same. Otherwise doing math with them doesn't make sense.
but as it is right now `TokenAmount`, what you mention is responsibility of the caller. `TokenAmount` could have another field for the `address` too.

Reply:
Ah, sorry. Got confused. I though the `TokenAmount` also contains the token address. In that case indeed no additional check is required. 👌